internal/handler/job/events/events.go

```diff
@@ -0,0 +1,127 @@
+package events
+
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/olebedev/emitter"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/siemens/wfx/generated/model"
+	"github.com/siemens/wfx/middleware/logging"
+)
+
+type FilterParams struct {
+	IDs       []string
+	ClientIDs []string
+	Workflows []string
+}
+
+type JobEvent struct {
+	Action Action     `json:"action"`
+	Job    *model.Job `json:"job"`
+}
+
+type Action string
+
+const (
+	ActionCreate           Action = "CREATE"
+	ActionDelete           Action = "DELETE"
+	ActionAddTags          Action = "ADD_TAGS"
+	ActionDeleteTags       Action = "DELETE_TAGS"
+	ActionUpdateStatus     Action = "UPDATE_STATUS"
+	ActionUpdateDefinition Action = "UPDATE_DEFINITION"
+)
+
+var e *emitter.Emitter = emitter.New(32)
+
+// AddSubscriber adds a new subscriber to receive job events filtered based on the provided filterParams.
+func AddSubscriber(ctx context.Context, filter FilterParams) (<-chan emitter.Event, error) {
+	log := logging.LoggerFromCtx(ctx)
+
+	// for logging purposes
+	subscriberID := uuid.New().String()
+
+	log.Info().
+		Str("subscriberID", subscriberID).
+		Dict("filterParams", zerolog.Dict().
+			Strs("clientIDs", filter.ClientIDs).
+			Strs("ids", filter.IDs).
+			Strs("workflows", filter.Workflows)).
+		Msg("Adding new subscriber for job events")
+
+	if mw := filter.toMiddleware(subscriberID); mw != nil {
+		return e.On("*", mw), nil
+	}
+	return e.On("*"), nil
+}
+
+// ShutdownSubscribers disconnects all subscribers.
+func ShutdownSubscribers() {
+	// note: e.Off("*") does not match topics containing the character '/'
```

Comment:
... since the library uses a glob and token are separated by `/`

Reply:
Added to the comment